src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -1224,10 +1227,6 @@ private async Task<IEnumerable<RestoreTargetGraph>> ExecuteLegacyRestoreAsync(
                 {
                     failed = true;
                 }
-                catch (AggregateException e) when (e.InnerException is FatalProtocolException)
```

Comment:
Can you help me understand why this is being done? 

This is regressing a previous improvement from https://github.com/NuGet/NuGet.Client/commit/52dd383ae0899680f487ff31e9dd63d0033eea32.

Reply:
A previous version of the prototype had bypassed async functionality by accessing the `.Result` property which was causing exceptions to be made into `AggregateExceptions`.  This was causing tests to fail as the logging code is catching stuff like `ProtocolException`.  In this code path, it wouldn't set `failed` to `true` causing unexpected behavior.

In the final version of the implementation, I have switched to using `.GetAwaiter().GetResult()` which doesn't aggregate exceptions so I think this is not needed anymore.

Reply:
I think I misread this code change. 
There's a FatalProtocolException catch right above it that still exists there. So it's not really regressing anything. 

Resolving.